puppet/modules/calico/templates/node-daemonset.yaml.erb

```diff
@@ -116,14 +97,30 @@ spec:
                   name: calico-config
                   key: etcd_cert
 <% end -%>
-            # Auto-detect the BGP IP address.
-            - name: IP
-              value: ""
-            # Setup a custom MTU value
-            - name: FELIX_IPINIPMTU
-              value: "<%= @mtu %>"
-            - name: FELIX_HEALTHENABLED
+<%- else -%>
+            # Use Kubernetes API as the backing datastore.
+            - name: DATASTORE_TYPE
+              value: "kubernetes"
+            # Wait for the datastore.
+            - name: WAIT_FOR_DATASTORE
+              value: "true"
+            # Enable IP-in-IP within Felix.
+            - name: FELIX_IPINIPENABLED
```

Comment:
I think the value here should be from `<%= @ipv4_pool_ipip_mode %>`

When you build a kubernetes backed calico, you should get that output:

```
kubectl get ippools default-ipv4-ippool -o yaml
[...]
spec:
  ipipMode: CrossSubnet
```
Currently it is `Always`